### DISCUSSION TITLE
RDKit Fingerprint aromatic bonds

### DISCUSSION THREAD

**Discussion by MherMatevosyan** [rdkit/rdkit#4417] [Q&A]
Does RDKFingerprint consider aromaticity? In image below I used RDKFingeprint hash to find substructures and it seems the fingerprint hash ignores the aromaticity of atoms/bonds involved. Is there a way to take this into account?

Also, if someone knows a better way to find for which molecules from a database (4 mln molecules) a given molecule is a superstructure of, I would really be thankful to know.
![image](https://user-images.githubusercontent.com/29042139/129354644-3d447f2c-78d2-488a-a986-ab27391406a7.png)

**Comment by bp-kelley** [comment]
I’m fairly certain the rdkit fingerprints handles automaticity.  If you are looking for substructures though, you need to use the pattern fingerprint in particular.

You will want to use

Chem.PatternFingerprint
And
DataStructs.AllProbeBitsMatch

see this blog for more details:

http://rdkit.blogspot.com/2013/11/fingerprint-based-substructure.html?m=1

If you are using Postgres, the rdkit cartridge does a good job of substructure searching.

alternatively, you can load the smiles into the SubstructLibrary and use the pattern holder which does all the work for you and handles multi-threading.
 
 more details on the blog:

http://rdkit.blogspot.com/2018/02/introducing-substructlibrary.html?m=1

good luck!

**Reply by MherMatevosyan** [reply]
Thanks! I will try to dig in the blogposts!
